### PR TITLE
🏗 Replace `globby` with `fast-glob`

### LINKED ISSUE #30223: 🐛 move off globby onto a lib with Windows support
Context: https://github.com/ampproject/amphtml/pull/30222.

Globby doesn't return windows paths.
We should either implement a global workaround or switch libs to one that explicitly supports windows.

### PR DESCRIPTION
**Pull request by rsimha**
We use `globby` to expand globs as part of various AMP dev-tool workflows. There are two problems with it:

1. It doesn't support Windows paths. See #30223.
2. It has [recently become](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) a pure ESM module, which doesn't work well with AMP's build system.

This PR replaces `globby` with `fast-glob`, which happens to be the underlying technology behind `globby`. This makes for a seamless replacement with no performance downsides.

Fixes #30223

### CODE REVIEW

File: build-system/tasks/lint.js [hunk 1]
```diff
@@ -123,11 +123,7 @@ function summarizeResults(results, fixedFiles) {
  * @return {Promise<void>}
  */
 async function lint() {
-  const filesToCheck = getFilesToCheck(
-    lintGlobs,
-    {gitignore: true},
-    '.eslintignore'
-  );
+  const filesToCheck = getFilesToCheck(lintGlobs, {}, '.eslintignore');
```
Comment (rcebulko) [inline build-system/tasks/lint.js:126]: Why the code change here?

File: build-system/tasks/check-exact-versions.js [hunk 2]
```diff
@@ -35,9 +35,9 @@ function check(file) {
  * @return {!Promise}
  */
 async function checkExactVersions() {
-  const packageJsonFiles = await globby([
+  const packageJsonFiles = await fastGlob([
     '**/package.json',
-    '!**/node_modules',
+    '!**/node_modules/**',
```
Comment (rcebulko) [inline build-system/tasks/check-exact-versions.js:40]: Why does this change?

**Review by rcebulko — APPROVED** [review]

File: build-system/tasks/lint.js (hunk 1 above)
Comment (rsimha) [reply build-system/tasks/lint.js:126]: `gitignore` is a `globby` add-on option, and not a native `fast-glob` option. In this case, I've already made sure that `.eslintignore` contains all the correct entries, and this code change is doing exactly what we want (and no more).

File: build-system/tasks/check-exact-versions.js (hunk 2 above)
Comment (rsimha) [reply build-system/tasks/check-exact-versions.js:40]: `fast-glob` requires this, while `globby` did not. We already use the convention across the rest of our globs, this was the one place where it was missing.

File: build-system/tasks/check-types.js [hunk 3]
```diff
@@ -204,7 +204,7 @@ async function typeCheck(targetName) {
 
   // If srcGlobs and externGlobs are defined, determine the externs/extraGlobs
   if (srcGlobs.length || externGlobs.length) {
-    opts.externs = externGlobs.flatMap(globby.sync);
```
Comment (samouri) [inline build-system/tasks/check-types.js:207]: unrelated note: we should go through all our usages of `globby.sync` and switch to async

File: build-system/test-configs/config.js [hunk 4]
```diff
@@ -113,13 +113,8 @@ const lintGlobs = [
 ];
 
 /**
-<<<<<<< HEAD
```
Comment (samouri) [inline build-system/test-configs/config.js:116]: ⭐

**Comment by samouri** [comment]
Thank you for the bugfix @rsimha!

File: build-system/tasks/check-types.js (hunk 3 above)
Comment (rcebulko) [reply build-system/tasks/check-types.js:207]: Same with file I/O, we do a looot of `readFileSync` etc iirc

**Pull request merged**